File: EloMark_laravel/app/Models/Course.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Course extends Model
{
    use HasFactory;

    protected $primaryKey = 'course_id'; // Important
    public $incrementing = true;
    protected $keyType = 'int';

    protected $fillable = ['course_name', 'course_code'];

    public function examMarks()
    {
        return $this->hasMany(ExamMark::class, 'course_id');
    }
}

File: elomark/lib/models/course.dart

class Course {
  final int courseId;
  final String courseName;
  final String courseCode;

  Course({
    required this.courseId,
    required this.courseName,
    required this.courseCode,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      courseId: json['course_id'],
      courseName: json['course_name'],
      courseCode: json['course_code'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'course_id': courseId,
      'course_name': courseName,
      'course_code': courseCode,
    };
  }
}